### PR TITLE
Fix LWJGL3 window not updating on resize. Resolves #6808

### LINKED ISSUE #6808: LWJGL3 windows no longer update while being resized
Looks like we have a regression on our hands. Videos are by @fgnm in #6785 - that issue is closed and the callback the issue it was about is resolved. While our windows used to update when resized, they don't anymore, which looks a bit odd.

My concern is more to do with games during development than once released, as resizing your window is an excellent way of telling if UI and all that is responding to different sizes correctly.

## 1.10.0

https://user-images.githubusercontent.com/60154347/156578877-accacbc5-fd5b-47ca-99f6-9b71627fcfb7.mp4

## 1.10.1-SNAPSHOT

https://user-images.githubusercontent.com/60154347/156578881-0845af51-4a3f-4233-9661-5710cf6e547d.mp4

## Affected platforms
- [ ] Android
- [ ] iOS
- [ ] HTML/GWT
- [X] Windows
- [ ] Linux
- [X] macOS

Windows and macOS tested by fgnm. Does not affect Linux in my test.

### PR DESCRIPTION
**Pull request by raeleus**
Since #6772, libGDX users have had an issue on the LWJGL3 platform when resizing the libGDX window on Windows. When players drag the window border, it does not call update. This results in black, unrendered areas until the user releases the mouse button. See #6808. This particularly impacts all tool developers who depend on UI reacting to window resize.

This fix, courtesy of @BambooBandit, addresses this problem. As noted in the issue, this only affects Windows users, so the fix should not be applied to Linux or Mac. Thank you.

### CODE REVIEW

**Comment by BambooBandit** [comment]
Yes, this change basically stops Windows from being pulled down under with Mac

**Comment by raeleus** [comment]
It was brought to my attention that Linux is actually affected by this issue, so I've included it with my fix. It is ready for review now. Thanks.

**Review by rafaskb — APPROVED** [review]

**Comment by crykn** [comment]
This PR looks great! However, I [still](https://github.com/libgdx/libgdx/issues/6808#issuecomment-1110198514) think, we should check whether the `postRunnable` code is even needed on macOS (or maybe only when used with the gdx-lwjgl3-glfw-awt-macos extension).

**Comment by HydrolienF** [comment]
`I have an old mac laptop laying around (2015 model) I can test with if that would help. What should I try out? Running this branch with and without the postRunnable code while testing the resizing?`
It seem's to be the rigth thing to test.
You can try to run master branch to compare with.

**Comment by crykn** [comment]
@JamesTKhan: Yes. I guess the simplest approach is to just include `SharedLibraryLoader.isMac` in [this](https://github.com/libgdx/libgdx/pull/7117/files#diff-040a0bcb5d798c1906e75ea04d512d8a2ce33b694b8b1c4fe8edd6e6ffd9a2c5R77) if-condition and then check whether it misbehaves with and without the gdx-lwjgl3-glfw-awt-macos extension.

**Comment by JamesTKhan** [comment]
I haven't forgotten about this, trying to test it today ran into some intellij version issues on the mac I gotta figure out.

**Comment by JamesTKhan** [comment]
Sorry guys, my Mac is apparently older then I thought. I cannot get latest libGDX to build on the Mac and on the current version of IntelliJ due to android and when I tried updating IntelliJ to a newer version it crashes on launch as IntelliJ is trying to make API calls this old OS does not have available

File: backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Graphics.java [hunk 1]
```diff
@@ -73,23 +74,35 @@ public class Lwjgl3Graphics extends AbstractGraphics implements Disposable {
...
-					updateFramebufferInfo();
-					if (!window.isListenerInitialized()) {
-						return;
-					}
-					window.makeCurrent();
-					gl20.glViewport(0, 0, backBufferWidth, backBufferHeight);
-					window.getListener().resize(getWidth(), getHeight());
-					window.getListener().render();
-					GLFW.glfwSwapBuffers(windowHandle);
+			if (SharedLibraryLoader.isWindows || SharedLibraryLoader.isLinux) {
```
Comment (crykn) [inline backends/gdx-backend-lwjgl3/src/com/badlogic/gdx/backends/lwjgl3/Lwjgl3Graphics.java:77]: I think we should replace this with `Configuration.GLFW_CHECK_THREAD0.get(true)` as @piotr-j suggested.

**Comment by raeleus** [comment]
It seems I missed the conversation since Piotr's comment. I've implemented the suggested changes, however it seems I managed to break the build. Can someone help me out? I don't understand what has changed.

**Comment by Berstanio** [comment]
> It seems I missed the conversation since Piotr's comment. I've implemented the suggested changes, however it seems I managed to break the build. Can someone help me out? I don't understand what has changed.

```
import org.lwjgl.opengl.GL32;
```
The import was added, but I think, it isn't needed.

**Comment by raeleus** [comment]
Oh, I see some changes have been made to this file since I've made the PR. Am I supposed to pull origin? I thought that was frowned upon.

**Comment by Berstanio** [comment]
I think you can just remove the import? This whole thing will be probably squashed merged in any case.

**Comment by raeleus** [comment]
Gah, Github Desktop just fetched the origin anyway. Oh well. Looks like it's ready for testing right now everyone. Thank you.

**Review by crykn — APPROVED** [review]

**Comment by tommyettinger** [comment]
I'd like to review this, but I only have a Windows computer that I can easily test on. Has Mac been tested enough that we think it's good?

**Review by Berstanio — APPROVED** [review]
Works fine for me on mac M1 ventura, without glfw-awt-macos extension.
(It crashes with glfw-awt-macos, https://github.com/libgdx/libgdx/issues/6992)

**Pull request merged**